src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -55,6 +109,40 @@ export module DotNet {
     return invokePossibleInstanceMethodAsync(assemblyName, methodIdentifier, null, args);
   }
 
+  /**
+   * Creates a JavaScript object reference that can be passed to .NET via interop calls.
+   * 
+   * @param jsObject The JavaScript Object used to create the JavaScript object reference.
+   * @returns The JavaScript object reference (this will be the same instance as the given object).
+   * @throws Error if the given value is not an Object.
+   */
+  export function createJSObjectReference(jsObject: any): any {
+    if (jsObject instanceof Object) {
+      if (!jsObject.hasOwnProperty(JSObject.ID_KEY) || cachedJSObjectsById[nextJsObjectId] !== jsObject) {
```

Comment:
Why are we checking `cachedJSObjectsById[nextJsObjectId] !== jsObject`? I don't get why we're looking for an entry with key `nextJsObjectId` - doesn't that represent the next key that we haven't yet used?

Reply:
I think this was a mistake, it was meant to say `cachedJSObjectsById[jsObject[JSObject.ID_KEY]] !== jsObject`. The intent was to catch the case in which an object was created by copying one used as a `JSObjectReference`.

Reply:
However, the logic will change anyway if we don't mutate the object instance.